%% LyX 2.2.1 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
%\VignetteIndexEntry{An introduction to m6A-express}
%\VignetteEncoding{UTF-8}
%\VignetteEngine{knitr::knitr}
%\VignetteKeywords{m6A-express,gene expression,RNA methylation,m6A, regulation}
%\VignettePackage{m6A-expression}
\documentclass{article}
\usepackage[sc]{mathpazo}
\usepackage[T1]{fontenc}
\usepackage{hyperref}

\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\textit{#1}}}
\newcommand{\Rmethod}[1]{{\texttt{#1}}}
\newcommand{\Rfunarg}[1]{{\texttt{#1}}}
\newcommand{\Rclass}[1]{{\textit{#1}}}
\newcommand{\Rcode}[1]{{\texttt{#1}}}

\newcommand{\software}[1]{\textsf{#1}}
\newcommand{\R}{\software{R}}
\newcommand{\m6A-express}{\Rpackage{m6A-express}}
\newcommand{\bam}{\texttt{BAM}

\title{An Introduction to \Rpackage{m6A-express}}
\author{Teng Zhang, PhD student}
\data {Modified: 10 Aug, 2020.}

\begin{document}
\SweaveOpts{concordance=TRUE}
\maketitle

<<options,echo=FALSE>>=
options (width=60)
@

\section{Introduction}
The \m6A-express{} R-package has been developed to predict genes whose expressions are regulaed by m6A methylation based on MeRIP-seq samples. \m6A-express{} is based on a negative binomial regression model with pooling prior information across genes to explicitly estimate the sample-to-sample variability between biological replicates and improve the precision and robustness of regression coefficients. The inputs of main function \Rpackage{m6A_Express} are RNA-seq data BAM files or INPUT BAM files of MeRIP-seq data, which are used to quantify gene expression, and IP BAM files with paired INPUT BAM files that are used to quantify methylation inensity in \m6A-express{} model. In addition, users need to provide the origianl gene annotation file. The \m6A-express{} package can be used in two scenarios:
\begin{itemize}
  \item Predict m6A regulated expression genes under multiple conditions, such as in mutiple tissues context, and we call it basic mode.
  \item Predict the differential methylation peak sites regulated differential expression genes in specific context and we call it DE-DM mode.
\end{itemize}
Gene annotation can be provided as a GTF file, a \Rclass{TxDb} object, or automatically downloaded from UCSC through the internet. The \m6A-express{} R-package can predict m6A regulated expression (m6A-reg-exp) genes for different species, such as human, mouse and yeast. 

We will in the next see how the two main functions can be accomplished in a single command. 

\section{Basic Mode}
Let us firstly load the package and get the toy data (came with the package) ready.

<<input bams files>>=
library("m6A-express")
gtf <- system.file("extdata", "example.gtf", package="m6A-express")
f1 <- system.file("extdata", "IP1.bam", package="m6A-express")
f2 <- system.file("extdata", "IP2.bam", package="m6A-express")
f3 <- system.file("extdata", "IP3.bam", package="m6A-express")
f4 <- system.file("extdata", "IP4.bam", package="m6A-express")
f5 <- system.file("extdata", "Input1.bam", package="m6A-express")
f6 <- system.file("extdata", "Input2.bam", package="m6A-express")
f7 <- system.file("extdata", "Input3.bam", package="m6A-express")
f8 <- system.file("extdata", "Input4.bam", package="m6A-express")
@ 

In the first scenario, \Rfunction {m6A-express} R-pacakge is to predict m6A-reg-exp genes in one condition or multiple conditions. The input data includes the gene annotation GTF file, gene expression data (Input samples of MeRIP-seq data), IP and Input samples of MeRIP-seq data in BAM format.
<<Basic Mode>>==
m6A_reg_expr_gene <- m6A_Express(express_data=INPUT_BAM, IP_BAM=IP_BAM, INPUT_BAM=INPUT_BAM, 
                                 annot_type="hg19", GENE_ANNO_GTF=gtf, pvalue=0.05,mode="basic")
@
The results will be saved in the specified output directory, including the m6A peak calling result by \Rfunction {exomePeak} R-package, the paired reads count and methylation intensity for each gene in tab-delimited text file and m6A-reg-expr genes in .XLS (tab-delimited) format file.

The peak calling result are detailedly explained in \Rfunction {exomePeak} R-package. The first column in tab-delimited text file is the given gene name and other columns show the paired reads count and methylation intensity of each gene. The prediction result of \Rfunction {m6A-express} includes five columns given the gene name, Beta0, Beta1, pvalue and FDR: 
\begin{itemize}
  \item gene name - The name of m6A methylation regulated expression gene in the given annotation file.  
  \item Beta0 - Beta0 is a gene-specific intercept and models the baseline log gene expression.
  \item Beta1 - Beta1 captures the influence of m6A methylation on gene expression.
  \item pvalue - pvalue of the predicted m6A-reg-exp genes.
  \item FDR - FDR of the predicted m6A-reg-exp genes.
\end{itemize}
